metrics_handler/README.md

```diff
@@ -0,0 +1,132 @@
+2. `gcloud functions deploy metrics_handler --runtime python37 --trigger-topic=begin-metrics-handler --memory=1024MB --entry-point=run_main --timeout=500s`
+3. `gcloud scheduler jobs create pubsub metrics-handler --schedule="*/15 * * * *" --topic=begin-metrics-handler --message-body="{}" --description="Kicks off the metric handler"`
+
+
+## Config
+
+The behavior of the metrics handler is configurable on a per-test basis. See [examples/metrics](examples/metrics) for some example config files, which are written in JSON. These are loaded into your test config via the `METRIC_CONFIG` key as shown in [examples/pt-cifar-tpu-v3-8.yaml](examples/pt-cifar-tpu-v3-8.yaml)
+
+There are 2 areas to configure:
+1. `metric_collection_config` controls which training metrics are collected when your test runs and how those metrics are aggregated.
+2. `regression_test_config` controls alerting and error reporting for each metric.
+
+
+#### metric_collection_config
+
+By default, the metrics handler will collect all metrics that your test writes to Tensorboard and will aggregate by using the final value for each metric. It will write these values to BigQuery.
+
+Full config options:
+
+```
+"metric_collection_config": {
+  # (Required) How to choose the value of each metric written to BigQuery.
+  # For each strategy, one row will be written per test run and the metric
+  # name will be e.g. accuracy_final or accuracy_max. Available strategies:
+  # "final": Save the final value for that metric.
+  # "max": Save the max value for that metric.
+  # "min": Save the min value for that metric.
+  "default_aggregation_strategies": ["final"],
+
+  # (Optional) Apply special aggregation strategies to some metrics. Key
+  # is the Tensorboard tag of the metric that your model writes during
+  # training and value is a list of strategies as described above.
+  "metric_to_aggregation_strategy": {
+    "loss": ["final", "min"],  # Save final loss and min loss.
+    "accuracy": ["max"],       # Save max accuracy only.
+  }
+
+  # (Optional) Ignore metrics with these Tensorboard tags. These metrics will
+  # not be aggregated or written to Bigquery and the metrics handler will not
+  # look for regressions in these metrics.
+  "tags_to_ignore": ["LearningRate"],
+
+  # (Optional) Compute the wall time between the test starting and the value
+  # of a specified Tensorboard tag reaching a specified threshold. If set,
+  # the metrics handler will write a row with metric name `time_to_accuracy`
+  # for each test run or send an alert if the threshold is never reached.
+  "time_to_accuracy": {
+    "accuracy_threshold": 76.0,  # Compute time needed to reach 76% acc.
+    "accuracy_tag": "Accuracy/test",  # Tag used in the SummaryWriter for acc.
+  },
+
+  # (Optional) Defaults to True. Set to false to disable all Bigquery writes.
+  "write_to_bigquery": "True",
+}
+```
+
+
+#### regression_test_config
+
+The metrics handler can send alerts if any training metrics regress. These
```

Comment:
nit: you use a hard line break here instead of letting it wrap like the other paragraphs.

We should pick one style and stick with it. I don't have a strong preference either way.